atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -1851,13 +1851,12 @@ private void internalPutMetadataForTables(
         final Map<Cell, byte[]> newMetadata = Maps.newHashMap();
         final Collection<CfDef> updatedCfs = Lists.newArrayList();
         for (Entry<Cell, byte[]> entry : metadataRequestedForUpdate.entrySet()) {
-            Value val = persistedMetadata.get(entry.getKey());
-            if (val == null || !Arrays.equals(val.getContents(), entry.getValue())) {
+            if (newMetadataFound(persistedMetadata.get(entry.getKey()), entry.getValue())) {
                 newMetadata.put(entry.getKey(), entry.getValue());
-                updatedCfs.add(getCfForTable(
-                        tableReferenceFromBytes(entry.getKey().getRowName()),
-                        entry.getValue()));
             }
+            updatedCfs.add(getCfForTable(
+                    tableReferenceFromBytes(entry.getKey().getRowName()),
+                    entry.getValue()));
```

Comment:
Talked to @tpetracca offline, we reach this check post filtering tables with changed metadata or changed cfdef. gc_grace_seconds update only appears as part of the cfdef and not metadata, hence this check filters it out. To avoid this confusion:
1. We can add another check here for change in cfdef.
2. Or Eliminate any checks in putMetadataInternal, ofc after ensuring all code paths filter the tables before calling this method.

Also, I will add ETE tests to ensure only required tables are updated, not all.